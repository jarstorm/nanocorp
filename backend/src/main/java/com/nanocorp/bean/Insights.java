package com.nanocorp.bean;

public class Insights {

	private String impressions;
	private String clicks;
	private String nanos_score;
	private String cost_per_click;
	private String click_through_rate;
	private String advanced_kpi_1;
	private String advanced_kpi_2;
	private String website_visits;

	public String getImpressions() {
		return impressions;
	}

	public void setImpressions(String impressions) {
		this.impressions = impressions;
	}

	public String getClicks() {
		return clicks;
	}

	public void setClicks(String clicks) {
		this.clicks = clicks;
	}

	public String getNanos_score() {
		return nanos_score;
	}

	public void setNanos_score(String nanos_score) {
		this.nanos_score = nanos_score;
	}

	public String getCost_per_click() {
		return cost_per_click;
	}

	public void setCost_per_click(String cost_per_click) {
		this.cost_per_click = cost_per_click;
	}

	public String getClick_through_rate() {
		return click_through_rate;
	}

	public void setClick_through_rate(String click_through_rate) {
		this.click_through_rate = click_through_rate;
	}

	public String getAdvanced_kpi_1() {
		return advanced_kpi_1;
	}

	public void setAdvanced_kpi_1(String advanced_kpi_1) {
		this.advanced_kpi_1 = advanced_kpi_1;
	}

	public String getAdvanced_kpi_2() {
		return advanced_kpi_2;
	}

	public void setAdvanced_kpi_2(String advanced_kpi_2) {
		this.advanced_kpi_2 = advanced_kpi_2;
	}

	public String getWebsite_visits() {
		return website_visits;
	}

	public void setWebsite_visits(String website_visits) {
		this.website_visits = website_visits;
	}
}
