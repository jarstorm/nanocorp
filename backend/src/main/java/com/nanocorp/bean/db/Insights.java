package com.nanocorp.bean.db;

import java.io.Serializable;

import javax.persistence.*;

@Entity
@Table(name = "NC_INSIGHTS")
public class Insights implements Serializable {

	private static final long serialVersionUID = -7974633271634453615L;

	/**
	 * Insights id
	 */
	@Id
	@Column(name = "I_ID")
	@GeneratedValue(strategy = GenerationType.AUTO)
	private long id;

	@Column(name = "I_IMPRESSIONS")
	private long impressions;

	@Column(name = "I_CLICKS")
	private long clicks;

	@Column(name = "I_WEBSITE_VISITS")
	private long websiteVisits;

	@Column(name = "I_NANOS_SCORE")
	private float nanosScore;

	@Column(name = "I_COST_CLICK")
	private float costPerClick;

	@Column(name = "I_CLICK_RATE")
	private float clickThroughRate;

	@Column(name = "I_KPI_1")
	private float kpi1;

	@Column(name = "I_KPI_2")
	private float kpi2;

	@OneToOne(mappedBy="insights")
	@PrimaryKeyJoinColumn
	private Platform platform;

	public long getId() {
		return id;
	}

	public void setId(long id) {
		this.id = id;
	}

	public long getImpressions() {
		return impressions;
	}

	public void setImpressions(long impressions) {
		this.impressions = impressions;
	}

	public long getClicks() {
		return clicks;
	}

	public void setClicks(long clicks) {
		this.clicks = clicks;
	}

	public long getWebsiteVisits() {
		return websiteVisits;
	}

	public void setWebsiteVisits(long websiteVisits) {
		this.websiteVisits = websiteVisits;
	}

	public float getNanosScore() {
		return nanosScore;
	}

	public void setNanosScore(float nanosScore) {
		this.nanosScore = nanosScore;
	}

	public float getCostPerClick() {
		return costPerClick;
	}

	public void setCostPerClick(float costPerClick) {
		this.costPerClick = costPerClick;
	}

	public float getClickThroughRate() {
		return clickThroughRate;
	}

	public void setClickThroughRate(float clickThroughRate) {
		this.clickThroughRate = clickThroughRate;
	}

	public float getKpi1() {
		return kpi1;
	}

	public void setKpi1(float kpi1) {
		this.kpi1 = kpi1;
	}

	public float getKpi2() {
		return kpi2;
	}

	public void setKpi2(float kpi2) {
		this.kpi2 = kpi2;
	}

	public Platform getPlatform() {
		return platform;
	}

	public void setPlatform(Platform platform) {
		this.platform = platform;
	}
}
