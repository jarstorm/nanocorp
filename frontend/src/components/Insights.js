import React, {Component} from 'react';

export default class Insights extends Component {
  	render() {
	  	const {data} = this.props;
	  	if (data !== null) {
	    return (
	      <div>	  
	      	<p><b>Insights</b></p>    
	      	<p>Clicks: {data.clicks}</p>
	      	<p>Clicks through rate: {data.clicksThroughRate}</p>
	      	<p>Cost per click: {data.costPerClick}</p>
	      	<p>Impressions: {data.impressions}</p>
	      	<p>Kpi1: {data.kpi1}</p>
	      	<p>Kpi2: {data.kpi2}</p>
	      	<p>Nanos score: {data.nanosScore}</p>
	      	<p>Website visits: {data.websiteVisits}</p>
	      </div>
	    )
	} else {
		return (<div><p><b>Insights</b></p></div>)
	}
  	}
}
