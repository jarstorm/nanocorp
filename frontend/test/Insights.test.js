// Link.react.test.js
import React from 'react';
import Insights from '../src/components/Insights';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

test('Draw Insights component', () => {  
  const data = {clicks: '100', kpi1: '1.0'}
  const component = renderer.create(<Insights data={data} />,
  );
  let dialog = component.toJSON();
  expect(dialog).toMatchSnapshot();  
});
