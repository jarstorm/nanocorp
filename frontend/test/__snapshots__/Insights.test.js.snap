// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Draw Insights component 1`] = `
<div>
  <p>
    <b>
      Insights
    </b>
  </p>
  <p>
    Clicks: 
    100
  </p>
  <p>
    Clicks through rate: 
  </p>
  <p>
    Cost per click: 
  </p>
  <p>
    Impressions: 
  </p>
  <p>
    Kpi1: 
    1.0
  </p>
  <p>
    Kpi2: 
  </p>
  <p>
    Nanos score: 
  </p>
  <p>
    Website visits: 
  </p>
</div>
`;
